services/train_xgboost_ad_classifier.py

```diff
@@ -0,0 +1,88 @@
+import glob
+import os
+from datetime import datetime
+
+import joblib
+import numpy as np
+import pandas as pd
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics import classification_report
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.utils.class_weight import compute_class_weight
+from xgboost import XGBClassifier
+
+csv_files = glob.glob("data/whisper_labeled_*.csv")
+df_list = [pd.read_csv(file, header=None, names=["timestamp", "id", "text", "label"], quotechar='"') for file in csv_files]
+df = pd.concat(df_list, ignore_index=True)
+
+df = df.dropna(subset=["text", "label"])
+df["label"] = df["label"].astype(int)
+df["timestamp"] = pd.to_datetime(df["timestamp"])
+df["hour"] = df["timestamp"].dt.hour
+df["minute"] = df["timestamp"].dt.minute
+
+def is_irregular_ad(row):
+    minute = row["minute"]
+    return ((0 <= minute < 27) or (30 <= minute < 57)) and row["label"] == 1
+
+irregular_ads = df[df.apply(is_irregular_ad, axis=1)]
+
+boosted_ads = pd.concat([irregular_ads] * 2, ignore_index=True)
+
+df = pd.concat([df, boosted_ads], ignore_index=True)
+
+X_text = df["text"]
+X_time = df[["hour", "minute"]]
+y = df["label"]
+
+classes = np.array(sorted(df["label"].unique()))
+weights = compute_class_weight("balanced", classes=classes, y=y)
+class_weights = dict(zip(classes, weights))
+
+classes = np.array(sorted(df["label"].unique()))
+weights = compute_class_weight("balanced", classes=classes, y=y)
+class_weights = dict(zip(classes, weights))
```

Comment:
```suggestion
classes = np.array(sorted(df["label"].unique()))
weights = compute_class_weight("balanced", classes=classes, y=y)
class_weights = dict(zip(classes, weights))
```

`compute_class_weight` 관련 코드가 동일하게 두 번 작성된 것 같아요 :)
중복 부분을 제거하면 더 깔끔해질 것 같습니다!

Reply:
좋은 리뷰 감사드립니다! 해당 사항 반영했습니다.